### PR TITLE
Support for non-BMP code points

### PR DESCRIPTION
**Pull request by flying-kestrel**
This adds basic support for non-BMP code points, tested with several SMP code points.

The relevant changes:
- Add support for TTF cmap format 12 (and latent 13 support since it's nearly identical)
- Change cmap lookups to look for "full repertoire" cmaps before BMP-only cmaps
- Added merged font to resources/ for demo and a simple example with a few SMP code points.
- Remove non-BMP check in `util/utf8_to_utf32.h`.

Let me know what you think!

### CODE REVIEW

**Comment by tomolt** [comment]
Thank you so much! This is looking really good!

It'll take a bit more time for me to merge the changes, since I want to read through everything thoroughly.

GCC 10 throws some harmless warnings about some local vars potentially not being initialized,
but valgrind confirmed that this doesn't result in a bug. I can clean up those warnings after merging.

I'll keep you updated.

File: schrift.c [hunk 1]
```diff
@@ -843,13 +843,52 @@ cmap_fmt6(SFT_Font *font, uint_fast32_t table, SFT_UChar charCode, SFT_Glyph *gl
...
 
+static int
+cmap_fmt12_13(SFT_Font *font, uint_fast32_t table, SFT_UChar charCode, SFT_Glyph *glyph, int which)
+{
+	uint32_t len, numEntries;
+	uint_fast32_t i;
+
+	*glyph = 0;
+
+	if (!is_safe_offset(font, table, 4))
```
Comment (tomolt) [inline schrift.c:854]: **Buffer Overrun:** This line only checks whether `table` contains at least 4 bytes.
But since we read 4 bytes starting at offset 4, we need to check for at least 8.
Since the table header must be at least 16 bytes long, we may as well check for 16.

Comment (tomolt) [reply schrift.c:854]: Your `cmap_fmt12_13()` routine scans through the entries naively.
If there aren't too many entries, this is fine.
But since TrueType guarantees that the entries are sorted by ascending `startCharCode`, this can be optimized with a binary search.

**Comment by tomolt** [comment]
Your `cmap_fmt12_13()` routine scans through the entries naively.
If there aren't too many entries, this is fine.
But since TrueType guarantees that the entries are sorted by ascending `startCharCode`, this can be optimized with a binary search.

**Comment by tomolt** [comment]
**TL;DR:** Once you've fixed the buffer overrun, you're golden!

While you're at it, could you also change the PR to merge into the `develop` branch instead of the `master` branch?
That makes it easier for me.

**Comment by tomolt** [comment]
All right, I'll merge now.

Optimizations and such would be nice to have, but what you've made is perfectly usable as-is.
And BTW, thanks for going the extra mile getting some real measurements on the format 12 entry counts!

This is going into the develop branch for now, but I'll try to turn it into a proper tagged release as soon as possible!
